EIPS/eip-draft_dtype_alias.md

```diff
@@ -0,0 +1,86 @@
+---
+eip: <to be assigned>
+title: dType Alias Extension - Decentralized Type System
+author: Loredana Cirstea (@loredanacirstea), Christian Tzurcanu (@ctzurcanu)
+discussions-to: https://github.com/ethereum/EIPs/issues/2192
+status: Draft
+type: Standards Track
+category: ERC
+created: 2019-07-16
+requires: 1900, 2157
+---
+
+## Simple Summary
+
+The dType Alias is a system for providing human-readable resource identifiers to on-chain content, supporting any type of data.
+
+## Abstract
+
+The dType Alias is a system for providing human-readable resource identifiers to on-chain content. A resource identifier is based on the type of data (identifier provided by dType, EIP-1900) and the data content (identifier provided by dType Storage Contracts, EIP-2157). It is a universal way of addressing content, supporting any type of data.
+
+
+## Motivation
+
+There are standards that currently address the need for attaching human-readable identifiers to Ethereum accounts, such as [EIP-137](http://eips.ethereum.org/EIPS/eip-137). However, they do not address the content of smart contracts, which can contain heterogeneous data that belongs to various accounts. Having uniquely identified resources opens the way to creating both human and machine-readable protocols on top of Ethereum. It also provides a basis for protocols based on functional programming.
+
+Imagine a dType `Profile` type, with related storage data about user profiles. One could access such a profile using an alias for the data content: `Alice@Profile`. For a `PaymentChannel` type, Alice can refer to her channel with Bob with `AliceBob.PaymentChannel`.
+This alias system can be used off-chain, to replace the old DNS system with a deterministic and machine-readable way of displaying content, based on the dType type's metadata.
+
+## Specification
+
+We define an `Alias` registry contract, that keeps track of the human-readable identifiers for data content.
+
+```
+interface Alias {
+
+    event AliasSet(bytes32 dtypeIdentifier, string name, string separator, bytes32 indexed identifier);
+
+    function setAlias(bytes32 dtypeIdentifier, string memory name, string memory separator, bytes32 identifier, bytes memory signature) public;
```

Comment:
A few technical comments:
- `string memory separator` seems to have a high overhead if only used for a single character
- `public` doesn't seem to be valid on interfaces
- the description doesn't seem to be complete (Who can set aliases? Anyone?)
- a rationale explaining why considering ENS is a bad idea for aliases (ENS itself is a hiearchical naming system in the core)
- an explanation why cross-chain aliases are needed/useful (since it references chainid below)

Reply:
Good points. I will address them.
I am writing an extended article on this ERC and I wanted to be able to provide a link to it, so I had to rush it.
Thank you for your time.

Reply:
Do you plan to fix them in this PR?

Reply:
Yes, in the next 24h.

Reply:
Done in https://github.com/ethereum/EIPs/pull/2193/commits/b154935f390d07a92e20e3b0bba94a8dcf1e1e56.

The chain id is only used for replay attacks on `setAlias`. An attacker can use a previous testnet signature to change the owner's mainnet alias. In this case, `dtypeIdentifier`, `identifier`, `name`, `separator`, `nonce` have high chances of being the same.  `aliasAddress` can be the same cross-chains. So, we need a `chainId`.
What I am not sure about is whether I should add the signature spec or not to the ERC - I added it to explain the `bytes memory signature` parameter in `setAlias`.